### ISSUE TITLE
problem with 2.0.0 version

### ISSUE THREAD

**Issue by lafin** [tjunnone/npm-check-updates#109]
Problems. Version 2.0.0
My package.json
```json
{
  "dependencies": {
    "leaflet": "^1.0.0-beta"
  },
  "devDependencies": {
    "react-router": "^1.0.0-beta3"
  }
}
```
1.  ncu -ua after see error

    ```
    Unhandled rejection TypeError: Cannot read property 'leaflet' of undefined
        at /mnt/files/dev/npm-check-updates/lib/npm-check-updates.js:200:69
        at Object.filterObject (/mnt/files/dev/npm-check-updates/node_modules/cint/cint.js:474:7)
        at printUpgrades (/mnt/files/dev/npm-check-updates/lib/npm-check-updates.js:199:34)
        at /mnt/files/dev/npm-check-updates/lib/npm-check-updates.js:118:13
        at tryCatcher (/mnt/files/dev/npm-check-updates/node_modules/bluebird/js/main/util.js:26:23)
        at Promise._settlePromiseFromHandler (/mnt/files/dev/npm-check-updates/node_modules/bluebird/js/main/promise.js:501:31)
        at Promise._settlePromiseAt (/mnt/files/dev/npm-check-updates/node_modules/bluebird/js/main/promise.js:577:18)
        at Promise._settlePromises (/mnt/files/dev/npm-check-updates/node_modules/bluebird/js/main/promise.js:693:14)
        at Async._drainQueue (/mnt/files/dev/npm-check-updates/node_modules/bluebird/js/main/async.js:123:16)
        at Async._drainQueues (/mnt/files/dev/npm-check-updates/node_modules/bluebird/js/main/async.js:133:10)
        at Immediate.Async.drainQueues [as _onImmediate] (/mnt/files/dev/npm-check-updates/node_modules/bluebird/js/main/async.js:15:14)
        at processImmediate [as _immediateCallback] (timers.js:371:17)
    ```

2. I can't override https://github.com/tjunnone/npm-check-updates/blob/master/lib/npm-check-updates.js#L287-L288
3. ```"react-router": "^1.0.0-beta3"``` it's not last version. ncu said what ```All dependencies match the latest package versions :)```

**Comment by metaraine** [comment]
Thanks, I'm looking into it now.

*It is much easier to track separate problems if they are added as separate github issues.*

**Comment by metaraine** [comment]
1. Yes, this appears to be a bug in npm-check-updates, as npm handles `^1.0.0-beta` fine. It looks like you can leave off the `.1`, but ncu doesn't handle that. You may want to change the dependency to `^1.0.0-beta.1` until this is fixed. I will track this issue in #111 from here on out.

2. Fixed! That was... definitely a problem.

3. This behavior is expected; let me explain. `npm install` and `npm install@latest` commands actually look for the version published on the "latest" tag. betas and prereleases can be published on other arbitrary tags, which can be installed with `npm install@beta` etc. Here are `react-router`'s published tags:

    ```sh
    $ npm view react-router dist-tags
    { latest: '0.13.3', alpha: '1.0.0-alpha2', beta: '1.0.0-beta3' }
    ```

    The *latest* published version is `0.13.3`, so that will be used by default. The *greatest* published version is `1.0.0-beta3`, available on the beta tag.

   In order to make npm-check-updates check the *greatest* (highest numbered) version available, simply add the `-t/--greatest` option:

    ```sh
    $ ncu --greatest
    ```

Closing this issue since the first issues is tracked in #111 and the others are fixed/explained. Thanks!
